### PR TITLE
compiler,reflect: use two bits of the meta byte for comparable/isBinary

### LINKED ISSUE #3683: Use two bits of type meta byte for Comparable / IsBinary
These two queries still require looping over all the fields in a struct, recursively.  Caching them in the meta byte would make both these operations `O(1)`.

### PR DESCRIPTION
**Pull request by dgryski**
Fixes #3683

### CODE REVIEW

**Comment by dgryski** [comment]
```
~/go/src/github.com/tinygo-org/tinygo/src/reflect $ tinygo test reflect
panic: reflect.Value.MapIndex: incompatible types for key
FAIL	reflect	0.454s
```

Edit: Looks like this just needed a `tinygo clean`..

File: src/reflect/type.go [hunk 1]
```diff
@@ -395,8 +395,10 @@ type Type interface {
 
 // Constants for the 'meta' byte.
 const (
-	kindMask  = 31 // mask to apply to the meta byte to get the Kind value
-	flagNamed = 32 // flag that is set if this is a named type
+	kindMask       = 31  // mask to apply to the meta byte to get the Kind value
+	flagNamed      = 32  // flag that is set if this is a named type
+	flagComparable = 64  // flag that is set if this type is comparable
+	flagIsBinary   = 128 // flag that is set if this type uses the hashmap binary algorithm
```
Comment (aykevl) [inline src/reflect/type.go:401]: Perhaps `1 << 5` etc would have been more readable.
I'm fine with it either way.

**Review by aykevl — APPROVED** [review]
LGTM. I kind of expected a small size reduction, but I guess those minimal examples that are tested don't include these reflect functions.

**Comment by deadprogram** [comment]
Thank you for the improvement @dgryski and to @aykevl for review. Now merging!

**Pull request merged**